common/src/main/java/io/druid/metadata/DefaultPasswordProvider.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Druid - a distributed column store.
+ * Copyright 2015 - Yahoo! Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.druid.metadata;
+
+import java.util.Map;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.metamx.common.ISE;
+
+
+public class DefaultPasswordProvider implements PasswordProvider
+{
+
+  private final static String PASSWORD_KEY = "password";
+
+  private final String password;
+
+  @JsonCreator
+  public DefaultPasswordProvider(Object spec)
+  {
+    password = convertValue(spec);
+  }
+
+  private static String convertValue(Object spec) {
+    if(spec instanceof String) {
+      return spec.toString();
+    } else if(spec instanceof Map && ((Map)spec).containsKey(PASSWORD_KEY)) {
```

Comment:
supporting string is required for backward compatibility. For example, currently users would have their configuration like following...
druid.metadata.storage.connector.password=diurd
this will stop working if we removed String support.

Additionally, map allows the new way of specifying same password, for example like
druid.metadata.storage.connector.password={"type":"default","password":"diurd"}
this is not required strictly but nice for consistency and a ref on how it would be if you wrote an extension.